domain/product/entity/entity.go

```diff
@@ -45,6 +45,22 @@ type Product struct {
 	ProductTypeName     string          `gorm:"-"`
 	ProductCategoryID   int64           `gorm:"-"`
 	ProductCategoryName string          `gorm:"-"`
+	Categories          []*Category     `gorm:"many2many:product_category_relations;"`
+}
+
+type Category struct {
+	base_model.BaseMasterDataModel
+	Name     string    `gorm:"type:varchar(255);not null;unique"`
+	ParentID int64     `gorm:"type:int;null"`
+	IsActive bool      `gorm:"type:bool;not null"`
+	Products []Product `gorm:"many2many:product_category_relations;"`
+}
+
+type ProductCategoryRelation struct {
+	ProductID  uuid.UUID `gorm:"type:uuid;not null"`
+	Product    Product   `gorm:"foreignKey:ProductID;references:ID"`
+	CategoryID int64     `gorm:"type:int;not null"`
```

Comment:
for the type, I see the differences between ProductID and CategoryID (use uuid & int), could we make it all as uuid?

Reply:
I set this way based on the existing data mas, should we change it accordingly?

Reply:
please make sure mas with others. 
I think It's better to have same type for the ID and I prefer UUID.

Reply:
I agree to add UUID for the category, since the behavior of category for this new table is different with the previous one, I think it's better to have UUID instead of id.

Reply:
updated mas, thanks